staker/l1_validator.go

```diff
@@ -265,7 +265,7 @@ func (v *L1Validator) generateNodeAction(
 	}
 
 	caughtUp, startCount, err := GlobalStateToMsgCount(v.inboxTracker, v.txStreamer, startState.GlobalState)
-	if v.blockValidator != nil && v.blockValidator.EspressoDebugging(startState.GlobalState.HotShotHeight) {
+	if v.blockValidator != nil && v.blockValidator.EspressoDebugging(startState.GlobalState.Batch) {
```

Comment:
Sovereign sequencer doesn't modify the global states like before, so the hotshot height should always be 0.
The staker test in current E2E test is to tamper with the block hash and the hotshot commitment as soon as the hotshot height reaches a given value. It will end up a challenge between the good staker and bad staker. And this challenge will end up the `ReadHotShotCommitment` OSP.

As for the sovereign sequencer, it seems not easy for us to build a challenge like this. At least I think we should do it another way.

Reply:
Thanks @ImJeremyHe, I have removed the staker test